### PR TITLE
Added support for PDO exception mode, which is default in PHP 8

### LINKED ISSUE #51: PHP 8 support - error handling mode has been changed from "silent" to "exceptions"
In PHP 8, the error handling mode has been changed from "silent" to "exceptions" - this is an issue in the condition which is built to handle return values instead of exceptions: https://github.com/mindplay-dk/sql/blob/0.7.6/src/framework/pdo/PreparedPDOStatement.php#L107

> The default error handling mode has been changed from "silent" to "exceptions". See [Errors and error handling](https://www.php.net/manual/en/pdo.error-handling.php) for details.

https://www.php.net/manual/en/migration80.incompatible.php#migration80.incompatible.pdo

It may support both configurations?

### PR DESCRIPTION
**Pull request by boan-jfm**
Added support for [PDO error mode set to `ERRMODE_EXCEPTION`](https://www.php.net/manual/en/pdo.error-handling.php), which is the new default in PHP 8. 

Added return type to the `getIterator` method within the `Result` model extending the `IteratorAggregate`. This has added return types in new PHP versions.

Resolves #51

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,15 +21,15 @@
...
+        "php": ">=7.0",
         "ext-pdo": "*",
         "mindplay/unbox": "^2",
         "psr/log": "^1"
     },
     "require-dev": {
         "mindplay/testies": "^0.3.2",
         "mindplay/benchpress": "^0.1",
-        "mockery/mockery": "^0.9",
+        "mockery/mockery": "^1.5.0",
```
Comment (boan-jfm) [inline composer.json:32]: This supports newer versions of PHP

File: src/framework/Result.php [hunk 2]
```diff
@@ -85,7 +85,7 @@ public function all()
      *
      * @return Iterator
      */
-    public function getIterator()
+    public function getIterator(): Iterator
```
Comment (boan-jfm) [inline src/framework/Result.php:88]: Must adhere to the updated interface in new PHP versions

**Comment by boan-jfm** [comment]
Seems like there are some bootstrapping issues with travis:

**SSL certificate failing**
> SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
> -- https://app.travis-ci.com/github/mindplay-dk/sql/jobs/575628921

**Environment not found**
>curl: (22) The requested URL returned error: 404
> -- https://app.travis-ci.com/github/mindplay-dk/sql/jobs/575628925

**Comment by mindplay-dk** [comment]
Thanks, but I'm not currently maintaining this.

If you need these changes anytime soon, you should fork.

**Comment by boan-jfm** [comment]
Okay, thank you for the reply 👍 😊

File: src/framework/pdo/PreparedPDOStatement.php [hunk 3]
```diff
@@ -104,7 +105,13 @@ public function execute()
     {
         $microtime_begin = microtime(true);
 
-        if (@$this->handle->execute()) {
+        try {
+            $execution_successful = @$this->handle->execute();
```
Comment (mindplay-dk) [inline src/framework/pdo/PreparedPDOStatement.php:109]: We probably shouldn't `@` error-suppress in a `try/catch`?

**Comment by mindplay-dk** [comment]
I've done some maintenance today, the project should be in a better state for PHP 8 now.

I've reviewed the change - if you'd like to update this PR, I will merge it.

**Comment by mindplay-dk** [comment]
Fixed in the next release

**Pull request closed without merging**